fdbserver/RestoreLoader.actor.cpp

```diff
@@ -442,10 +443,11 @@ void _parseSerializedMutation(VersionedMutationsMap* pkvOps, SerializedMutationL
 
 		StringRefReaderMX vReader(val, restore_corrupted_data());
 		vReader.consume<uint64_t>(); // Consume the includeVersion
-		uint32_t val_length_decoded =
-		    vReader.consume<uint32_t>(); // Parse little endian value, confirmed it is correct!
-		ASSERT(val_length_decoded ==
-		       val.size() - 12); // 12 is the length of [includeVersion:uint64_t][val_length:uint32_t]
+		// TODO(xumengpanda): verify the protocol version is compatible and raise error if needed
```

Comment:
Was this TODO item meant to be left in?

Reply:
Yes for the TODO.  The performant restore currently assumes the backup and restore are in the same version of code. This TODO wants to at least report a meaningful error when the backup protocol version changes (e.g., when we upgrade the backup to the new backup @jzhou77  is working on.